### PR TITLE
Baip 277/add transparent module

### PR DESCRIPTION
**Pull request by Mayt34**
Add transparent module to Dockerfile

## Summary by CodeRabbit

- **New Features**
	- Introduced a new build stage for a transparent module in the Dockerfile, enhancing the overall build process.
	- Integrated the transparent module into the final image, ensuring it is readily available for use.
	- Updated the version of the ModuleGateway project to 1.2.1, ensuring users benefit from the latest improvements.

These changes streamline the deployment process and improve the functionality of the application.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -62,8 +62,18 @@ RUN cmake -DCMAKE_BUILD_TYPE=Release -DBRINGAUTO_INSTALL=ON -DCMAKE_INSTALL_PREF
...
 
+ARG TRANSPARENT_MODULE_VERSION=v1.0.1
 
+WORKDIR /home/bringauto/
+ADD --chown=bringauto:bringauto https://github.com/bringauto/transparent-module.git#$TRANSPARENT_MODULE_VERSION transparent-module
+
+WORKDIR /home/bringauto/transparent-module/_build
+RUN cmake .. -DCMAKE_BUILD_TYPE=Release -DBRINGAUTO_INSTALL=ON -DCMAKE_INSTALL_PREFIX=/home/bringauto/modules/transparent_module/ \
+    -DFLEET_PROTOCOL_BUILD_EXTERNAL_SERVER=OFF .. && \
+    make install
```
[the remark this answers is not in the document]
Comment (Mayt34) [reply Dockerfile]: Fized

**Review by MarioIvancik — CHANGES_REQUESTED** [review]
Few changes are needed:
- module gateway version in CMakeLists needs to be incremented
- transparent module needs to be added to relevant config files in the resources folder

File: Dockerfile [hunk 2]
```diff
@@ -62,8 +62,18 @@ RUN cmake -DCMAKE_BUILD_TYPE=Release -DBRINGAUTO_INSTALL=ON -DCMAKE_INSTALL_PREF
     -DFLEET_PROTOCOL_BUILD_EXTERNAL_SERVER=OFF .. && \
     make install
 
+FROM bringauto/cpp-build-environment:latest AS transparent_module_builder
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply Dockerfile:65]: The latest tag is require by our CI configuration

File: Dockerfile [hunk 3]
```diff
@@ -62,8 +62,18 @@ RUN cmake -DCMAKE_BUILD_TYPE=Release -DBRINGAUTO_INSTALL=ON -DCMAKE_INSTALL_PREF
...
 
+ARG TRANSPARENT_MODULE_VERSION=v1.0.1
 
+WORKDIR /home/bringauto/
+ADD --chown=bringauto:bringauto https://github.com/bringauto/transparent-module.git#$TRANSPARENT_MODULE_VERSION transparent-module
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply Dockerfile:70]: The used way is required when frequently building the docker image during development. This way allows it to check for remote changes on github when rebuilding the image.

**Review by MarioIvancik — APPROVED** [review]

**Pull request merged**